### DISCUSSION TITLE
Issue : only able to retrieve 100 data in a `Read Data` operation in a flow

### DISCUSSION THREAD

**Discussion by bhagya1101** [directus/directus#17893] [Archive]
Hi Team,
I am creating a flow where the trigger is `Webhook` means it will trigger when the `/flow/trigger/flow_id` will have a `GET` call.
In that, I am performing `Read Data` operation for a collection which has number of data around 250.
But when I am connecting it with other operation that is `run script`, I am returning the whole data of the collection.
Normally, when I do an `GET` call on `/flow/trigger/flow_id`, I should get all 250 data but I am getting only 100 data.

Do any of you have any idea that how can I get whole 250 data?
Thanks

**Comment by rijkvanzanten** [comment]
> I should get all 250 data but I am getting only 100 data.

The default value for the `limit` query option is `100` by default. Set the `limit` query param to something higher to get more data 👍🏻

**Reply by bhagya1101** [reply]
I am getting my data by doing an `GET` api call on flow API and I am unable to apply limit on flow API.

**Reply by rijkvanzanten** [reply]
I meant in the options for the "Read Data" operation 👍🏻

**Reply by bhagya1101** [reply]
How do I add limit to an operation in flow? Because I dont see any option to add limit in flows??

**Reply by abhijit-sethi-groww** [reply]
You should consider using the directus sdk instead, it's highly customisable and well documented.
[https://docs.directus.io/reference/items.html#get-items](url)

**Reply by rijkvanzanten** [reply]
> How do I add limit to an operation in flow? Because I dont see any option to add limit in flows??

There's a query option you can use to configure any query parameters:
<img width="412" alt="CleanShot 2023-03-27 at 10 10 40@2x" src="https://user-images.githubusercontent.com/9141017/227964827-e54c30cd-7965-44c7-a063-2346a17e1c30.png">

**Reply by bhagya1101** [reply]
Thanks a lot...

**Reply by ymys** [reply]
Hi, what maximum limit? I tried 10.000 got error 524. Tried 5.000 success but sometimes "Service api is unavailable. Under pressure."

Total data 200.000.
